### PR TITLE
update documentations of reflection, bridge and jsbBridgeWrapper

### PR DESCRIPTION
**Pull request by timlyeee**
Re: #https://github.com/cocos/cocos-engine/issues/11135

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [x] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/native-binding/index.ts [hunk 1]
```diff
@@ -891,7 +891,8 @@ export declare namespace native {
     export namespace reflection{
         /**
          * https://docs.cocos.com/creator/manual/zh/advanced-topics/java-reflection.html
-         * call OBJC/Java static methods
+         * @en call OBJC/Java static methods
```
Comment (minggo) [inline cocos/native-binding/index.ts]: it is better to use full name of `OBJC `.

**Comment by minggo** [comment]
@cocos-robot run test cases

File: cocos/native-binding/index.ts [hunk 2]
```diff
@@ -863,5 +863,133 @@ export declare namespace native {
...
+         * @zh 调用 Objective-C/Java 静态方法
+         *
+         * @param className
+         * @param methodName
+         * @param methodSignature
+         * @param parameters
+         */
+        export function callStaticMethod (methodName: string, methodSignature: string, ...parameters:any): any {
+            return undefined;
+        }
```
Comment (PPpro) [inline cocos/native-binding/index.ts]:
```suggestion
        export function callStaticMethod (methodName: string, methodSignature: string, ...parameters:any): any;
```

**Comment by PPpro** [comment]
- please remove the file `platforms/native/engine/jsb-bridge-wrapper.js`  `jsb-bridge.js` and `jsb-reflection.js`
- also we need to remove the relative require statement in `platforms/native/engine/index.js`
- please remove the type declaration in `engine/@types/jsb.d.ts` of  `reflection` `bridge`   `jsbBridgeWrapper`, we've migrated these type dedeclarations into `native-binding/index.ts`

File: cocos/native-binding/impl.ts [hunk 3]
```diff
@@ -1,4 +1,130 @@
...
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ */
+import { legacyCC } from "../core/global-exports";
```
Comment (PPpro) [inline cocos/native-binding/impl.ts]:
```suggestion
import { sys } from  '../core/platform/sys';
```

we can access `sys` by importing sys module

File: cocos/native-binding/index.ts [hunk 4]
```diff
@@ -863,5 +863,131 @@ export declare namespace native {
...
+     * ```
+     * native.bridge.onNative = (event, data) => {
+     *   if (event === 'send_message') {
+     *    console.log(data);
+     *  }
+     * }
+     * ```
+     * ```java
+     *  JsbBridge.sendToScript('send_message', 'hello world');
+     * ```
```
Comment (minggo) [inline cocos/native-binding/index.ts:898]: I am not sure if `java` works. So it is better to add comment `// Java codes`.

File: cocos/native-binding/index.ts [hunk 5]
```diff
@@ -863,5 +863,131 @@ export declare namespace native {
...
+    export namespace reflection {
+        /**
+         * https://docs.cocos.com/creator/manual/zh/advanced-topics/java-reflection.html
+         * @en call Objective-C/Java static methods
+         * @zh 调用 Objective-C/Java 静态方法
+         *
+         * @param className
+         * @param methodName
+         * @param methodSignature
+         * @param parameters
```
Comment (minggo) [inline cocos/native-binding/index.ts]: What's the meaning of the parameters?

**Review by PPpro — APPROVED** [review]

**Comment by timlyeee** [comment]
@cocos-robot run test cases

**Review by minggo — APPROVED** [review]

**Pull request merged**